brick/dockerlib.py

```diff
@@ -55,26 +55,24 @@ def docker_build(
         dockerfile_contents += f'\nLABEL brick.dependency_hash="{dependency_hash}"'
         images_matching_hash = get_image_names_with_dependency_hash(dependency_hash)
         logger.debug(
-            f"Found {len(images_matching_hash)} image(s) matching dependency hash ({images_matching_hash[0:2]}..)"
+            f"Found {len(images_matching_hash)} image(s) matching dependency hash {dependency_hash} ({images_matching_hash[0:5]}..)"
         )
 
         images_are_build = set(tags).issubset(set(images_matching_hash))
         if images_are_build:
             logger.debug(f"Skipping docker build as images are up to date with input dependencies")
             return tag_to_return
 
-        # Investigate if we can promote images instead of building them again
-        image_name = tag_to_return.split(":")[0]
+        # Investigate if we can promote an image instead of building it again
+        image_names = set([t.split(":")[0] for t in tags])
```

Comment:
Linting error here:

`/home/runner/work/brick/brick/brick/dockerlib.py:67: [R1718(consider-using-set-comprehension), docker_build] Consider using a set comprehension`

I'm not strong in Python, but this should fix it, right? 😄 
```suggestion
        image_names = {[t.split(":")[0] for t in tags]}
```

Reply:
Thanks! Who set up these strict linting rule!? :D